x-pack/legacy/plugins/ml/public/jobs/new_job_new/common/job_creator/util/general.ts

```diff
@@ -51,6 +57,64 @@ function getDetectors(job: Job, datafeed: Datafeed) {
         },
       ];
     }
+  } else {
+    // all other detectors.
+    detectors = detectors.map(d => {
+      // if a count function is used, add EVENT_RATE_FIELD_ID as its field
+      if (d.function === ML_JOB_AGGREGATION.COUNT) {
```

Comment:
nit: same as my other comment, maybe this could use `else if` to avoid running through all `if` checks. maybe even a case for `switch` since all checks are against the same `d.function`?

Reply:
counter nit... each `if` contains a `return`, so it'll break out once it his a match.

Reply:
yeah a switch is far nicer